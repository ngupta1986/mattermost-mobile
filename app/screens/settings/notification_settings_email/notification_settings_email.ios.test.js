// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {configure, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({adapter: new Adapter()});

import NotificationSettingsEmailIos from './notification_settings_email.ios.js';

jest.mock('app/utils/theme', () => {
    const original = require.requireActual('app/utils/theme');
    return {
        ...original,
        changeOpacity: jest.fn(),
    };
});

describe('NotificationSettingsEmailIos', () => {
    const baseProps = {
        currentUserId: 'current_user_id',
        emailInterval: '30',
        enableEmailBatching: false,
        navigator: {setOnNavigatorEvent: () => {}}, // eslint-disable-line no-empty-function
        actions: {
            savePreferences: () => {}, // eslint-disable-line no-empty-function
        },
        sendEmailNotifications: true,
        siteName: 'Mattermost',
        theme: {
            centerChannelBg: '#aaa',
            centerChannelColor: '#aaa',
        },
    };

    test('should match snapshot, renderEmailSection', () => {
        const wrapper = shallow(
            <NotificationSettingsEmailIos {...baseProps}/>
        );

        expect(wrapper.instance().renderEmailSection()).toMatchSnapshot();
    });

    test('should call saveEmailNotifyProps on onNavigatorEvent', () => {
        const wrapper = shallow(
            <NotificationSettingsEmailIos {...baseProps}/>
        );

        const instance = wrapper.instance();
        instance.saveEmailNotifyProps = jest.fn();
        instance.onNavigatorEvent({type: 'ScreenChangedEvent', id: 'willDisappear'});

        expect(instance.saveEmailNotifyProps).toHaveBeenCalledTimes(1);
    });

    test('should macth state on setEmailNotifications', () => {
        const wrapper = shallow(
            <NotificationSettingsEmailIos {...baseProps}/>
        );

        wrapper.setState({email: 'false', interval: '0'});
        wrapper.instance().setEmailNotifications('30');
        expect(wrapper.state({email: 'true', interval: '30'}));

        wrapper.instance().setEmailNotifications('0');
        expect(wrapper.state({email: 'false', interval: '0'}));

        wrapper.instance().setEmailNotifications('3600');
        expect(wrapper.state({email: 'true', interval: '3600'}));
    });

    test('should call props.actions.savePreferences on saveUserNotifyProps', () => {
        const props = {...baseProps, actions: {savePreferences: jest.fn()}};
        const wrapper = shallow(
            <NotificationSettingsEmailIos {...props}/>
        );

        wrapper.setState({email: 'true', interval: '3600'});
        wrapper.instance().saveEmailNotifyProps();
        expect(props.actions.savePreferences).toHaveBeenCalledTimes(1);
        expect(props.actions.savePreferences).toBeCalledWith(
            'current_user_id',
            [
                {category: 'notifications', name: 'email', user_id: 'current_user_id', value: 'true'},
                {category: 'notifications', name: 'email_interval', user_id: 'current_user_id', value: '3600'},
            ]
        );
    });
});
