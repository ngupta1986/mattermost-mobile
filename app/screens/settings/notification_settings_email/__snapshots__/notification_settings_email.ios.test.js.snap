// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationSettingsEmailIos should match snapshot, renderEmailSection 1`] = `
<section
  disableFooter={false}
  footerDefaultMessage="Email notifications are sent for mentions and direct messages when you are offline or away from {siteName} for more than 5 minutes."
  footerId="user.settings.notifications.emailInfo"
  footerValues={
    Object {
      "siteName": "Mattermost",
    }
  }
  headerDefaultMessage="SEND EMAIL NOTIFICATIONS"
  headerId="mobile.notification_settings.email.send"
  theme={
    Object {
      "centerChannelBg": "#aaa",
      "centerChannelColor": "#aaa",
    }
  }
>
  <Component>
    <sectionItem
      action={[Function]}
      actionType="select"
      actionValue="30"
      label={
        <InjectIntl(FormattedText)
          defaultMessage="Immediately"
          id="user.settings.notifications.email.immediately"
        />
      }
      selected={true}
      theme={
        Object {
          "centerChannelBg": "#aaa",
          "centerChannelColor": "#aaa",
        }
      }
    />
    <Component
      style={
        Object {
          "backgroundColor": undefined,
          "flex": 1,
          "height": 1,
          "marginLeft": 15,
        }
      }
    />
    <sectionItem
      action={[Function]}
      actionType="select"
      actionValue="0"
      label={
        <InjectIntl(FormattedText)
          defaultMessage="Never"
          id="user.settings.notifications.email.never"
        />
      }
      selected={false}
      theme={
        Object {
          "centerChannelBg": "#aaa",
          "centerChannelColor": "#aaa",
        }
      }
    />
  </Component>
</section>
`;
